test/parallel/test-http-client-req-error-dont-double-fire.js

```diff
@@ -0,0 +1,45 @@
+// Copyright Joyent, Inc. and other Node contributors.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a
+// copy of this software and associated documentation files (the
+// "Software"), to deal in the Software without restriction, including
+// without limitation the rights to use, copy, modify, merge, publish,
+// distribute, sublicense, and/or sell copies of the Software, and to permit
+// persons to whom the Software is furnished to do so, subject to the
+// following conditions:
+//
+// The above copyright notice and this permission notice shall be included
+// in all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS
+// OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+// MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN
+// NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM,
+// DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR
+// OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE
+// USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+'use strict';
+const common = require('../common');
+const assert = require('assert');
+const http = require('http');
+const url = require('url');
+const URL = url.URL;
+const testPath = '/foo?bar';
+
+const u = `http://${common.localhostIPv4}:40404${testPath}`;
+const req = http.get(u);
+let count = 0;
+req.on('error', common.mustCall((err) => {
+  count++;
+  console.log('req error emit: %s, count: %s', err, count);
+  throw new Error('mock unexpected code error');
+}));
+
+let uncaughtExceptionCount = 0;
+process.on('uncaughtException', common.mustCall((err) => {
+  uncaughtExceptionCount++;
+  console.log(' -> uncaughtException fire, count: %s', uncaughtExceptionCount);
+  assert(count === 1, 'http request error event should only fire once');
```

Comment:
These asserts aren't necessary since `common.mustCall()` already checks that the functions are called exactly the specified number of times (with 1 being the default).